Map.clear w/ GenericMapStore and default settings fails w/ small amount of data

<!--
Thanks for reporting your issue. Please share with us the following information, to help us resolve your issue quickly and efficiently.
-->
  
**Describe the bug**
When a map is configured with a `GenericMapStore`, and `map.clear()` is used, this operation can take longer than expected or even timeout completely on a small amount of data in the map (couple hundred entries). This is with the default Hikari CP settings. Errors observed in the logs are (full log attached):

Member:
```
Caused by: java.sql.SQLTransientConnectionException: HikariPool-1 - Connection is not available, request timed out after 30000ms.
        at com.hazelcast.com.zaxxer.hikari.pool.HikariPool.createTimeoutException(HikariPool.java:696)
        at com.hazelcast.com.zaxxer.hikari.pool.HikariPool.getConnection(HikariPool.java:197)
        at com.hazelcast.com.zaxxer.hikari.pool.HikariPool.getConnection(HikariPool.java:162)
        at com.hazelcast.com.zaxxer.hikari.HikariDataSource.getConnection(HikariDataSource.java:100)
        at com.hazelcast.datastore.impl.CloseableDataSource.getConnection(CloseableDataSource.java:53)
        at com.hazelcast.jet.sql.impl.connector.jdbc.SelectProcessorSupplier.lambda$get$d66573d5$1(SelectProcessorSupplier.java:80)
        at com.hazelcast.function.SupplierEx.get(SupplierEx.java:45)
        at com.hazelcast.jet.impl.connector.ReadJdbcP.init(ReadJdbcP.java:126)
...
```

Client (where `map.clear()` is executed):
```
Exception in thread "main" com.hazelcast.client.UndefinedErrorCodeException: Class name: com.hazelcast.sql.HazelcastSqlException, Message: com.hazelcast.jet.JetException: Execution on a member failed: com.hazelcast.jet.JetException: Exception in ProcessorTasklet{08d2-d01b-08ae-0001/Select (people)#0}: java.sql.SQLTransientConnectionException: HikariPool-1 - Connection is not available, request timed out after 30000ms.
        at com.hazelcast.client.UndefinedErrorCodeException.wrap(UndefinedErrorCodeException.java:64)
        at com.hazelcast.client.UndefinedErrorCodeException.wrap(UndefinedErrorCodeException.java:34)
        at com.hazelcast.spi.impl.operationservice.impl.InvocationFuture.returnOrThrowWithGetConventions(InvocationFuture.java:110)
        at com.hazelcast.client.impl.spi.impl.ClientInvocationFuture.resolveAndThrowIfException(ClientInvocationFuture.java:95)
        at com.hazelcast.client.impl.spi.impl.ClientInvocationFuture.resolveAndThrowIfException(ClientInvocationFuture.java:40)
        at com.hazelcast.spi.impl.AbstractInvocationFuture.get(AbstractInvocationFuture.java:617)
        at com.hazelcast.client.impl.spi.ClientProxy.invoke(ClientProxy.java:215)
        at com.hazelcast.client.impl.proxy.ClientMapProxy.clear(ClientMapProxy.java:1665)
        at com.qe.TimerClient.main(TimerClient.java:124)
```

Also note: `com.hazelcast.client.UndefinedErrorCodeException` doesn't look so good. This error should be defined, and a helpful hint may be provided to the user if possible.

**Expected behavior**
Should complete `map.clear()` in a "reasonable" amount of for a such a small data set (single-digit seconds or less). The loading of the same data happens much faster.

**To Reproduce**
Prerequisite: a database with an existing table that can be used with a `GenericMapStore`-backed `IMap`. In my setup, I'm using a Postgres docker container w/ port 5432 published to the host. HZ member and client are both running on the host machine directly.

Steps to reproduce the behavior:
1. Start a member with `external-data-store` and `map` configured similar to the below example
2. Load some data into the map. It can be from external store (i.e. `MapLoader.loadAll()`) or `IMap` put/set. In my case 200 entries caused the `clear()` to fail consistently. With smaller numbers, the issue was intermittent but the operation still takes long to complete, and HikariCP timeouts are still observed on the member side.
3. Run `IMap.clear()` from the client
4. Observe the timeout exceptions like those listed above

config:
```
    <map name="people">
        <map-store enabled="true">
            <class-name>com.hazelcast.mapstore.GenericMapStore</class-name> 
            <properties>
                <property name="external-data-store-ref">postgres</property> 
            </properties>
        </map-store>
    </map>
    <external-data-store name="postgres">
        <class-name>com.hazelcast.datastore.JdbcDataStoreFactory</class-name>
        <properties>
            <property name="jdbcUrl">jdbc:postgresql://127.0.0.1:5432/test</property>
            <property name="username">postgres</property>
            <property name="password">***</property>
        </properties>
    </external-data-store>
```

**Additional context**

[member.log](https://github.com/hazelcast/hazelcast/files/9839332/member.log)


<!--
Add any other context about the problem here.

Common details that we're often interested in:
- Detailed description of the steps to reproduce your issue
- Logs and stack traces, if available
- Hazelcast version that you use (e.g. 3.4, also specify whether it is a minor release or the latest snapshot)
- If available, integration module versions (e.g. Tomcat, Jetty, Spring, Hibernate). Also, include their detailed configuration information such as web.xml, Hibernate configuration and `context.xml` for Spring
- Cluster size, i.e. the number of Hazelcast cluster members
- Number of the clients
- Version of Java. It is also helpful to mention the JVM parameters
- Operating system. If it is Linux, kernel version is helpful
- Unit test with the `hazelcast.xml` file. If you could include a unit test which reproduces your issue, we would be grateful
-->

--- a/extensions/mapstore/src/test/java/com/hazelcast/mapstore/GenericMapStoreIntegrationTest.java
+++ b/extensions/mapstore/src/test/java/com/hazelcast/mapstore/GenericMapStoreIntegrationTest.java
@@ -32,6 +32,7 @@ import com.hazelcast.sql.HazelcastSqlException;
 import com.hazelcast.sql.impl.QueryException;
 import com.hazelcast.test.ExceptionRecorder;
 import com.hazelcast.test.HazelcastSerialClassRunner;
+import com.hazelcast.test.annotation.NightlyTest;
 import com.hazelcast.test.annotation.QuickTest;
 import com.hazelcast.test.jdbc.H2DatabaseProvider;
 import com.hazelcast.test.jdbc.TestDatabaseProvider;
@@ -89,7 +90,8 @@ public class GenericMapStoreIntegrationTest extends JdbcSqlTestSupport {
         databaseProvider = testDatabaseProvider;
         dbConnectionUrl = databaseProvider.createDatabase(JdbcSqlTestSupport.class.getName());
 
-        memberConfig = smallInstanceConfig()
+        // Do not use small config to run with more threads and partitions
+        memberConfig = new Config()
                 // Need to set filtering class loader so the members don't deserialize into class but into GenericRecord
                 .setClassLoader(new FilteringClassLoader(newArrayList("org.example"), null))
                 .addDataConnectionConfig(
@@ -97,6 +99,7 @@ public class GenericMapStoreIntegrationTest extends JdbcSqlTestSupport {
                                 .setType("jdbc")
                                 .setProperty("jdbcUrl", dbConnectionUrl)
                 );
+        memberConfig.getJetConfig().setEnabled(true);
 
         ClientConfig clientConfig = new ClientConfig();
 
@@ -387,6 +390,28 @@ public class GenericMapStoreIntegrationTest extends JdbcSqlTestSupport {
         Row row = new Row("__map-store." + tableName);
         List<Row> rows = Collections.singletonList(row);
         assertTrueEventually(() -> assertDoesNotContainRow(client, "SHOW MAPPINGS", rows), 30);
+    }
+
+    /**
+     * Regression test for https://github.com/hazelcast/hazelcast/issues/22567
+     */
+    @Test(timeout = 30_000L)
+    @Category(NightlyTest.class)
+    public void testClear() {
+        HazelcastInstance client = client();
+        IMap<Integer, Person> map = client.getMap(tableName);
+
+        for (int i = 0; i < 10; i++) {
+            logger.info("Iteration " + i);
+            Map<Integer, Person> toInsert = new HashMap<>();
+            for (int j = 0; j < 10_000; j++) {
+                toInsert.put(j, new Person(j, "name-" + j));
+            }
+            map.putAll(toInsert);
+            logger.info("Inserted 10k items");
+            map.clear();
+            logger.info("Cleared map");
+        }
     }
 
     @Test

--- a/hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/connector/jdbc/JdbcSqlConnector.java
+++ b/hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/connector/jdbc/JdbcSqlConnector.java
@@ -354,7 +354,7 @@ public class JdbcSqlConnector implements SqlConnector {
                         builder.query(),
                         table.getBatchLimit()
                 )
-        ));
+        ).localParallelism(1));
     }
 
     @Nonnull
@@ -501,7 +501,7 @@ public class JdbcSqlConnector implements SqlConnector {
                             upsertStatement,
                             jdbcTable.getBatchLimit()
                     )
-            );
+            ).localParallelism(1);
         }
         // Unsupported dialect. Create Vertex with the INSERT statement
         VertexWithInputConfig vertexWithInputConfig = insertProcessor(context);

--- a/hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/connector/jdbc/JdbcSqlConnector.java
+++ b/hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/connector/jdbc/JdbcSqlConnector.java
@@ -448,27 +448,27 @@ public class JdbcSqlConnector implements SqlConnector {
     ) {
 
         if (!hasInput) {
-            // There is no input, we push the whole update query to the database, but we need single dummy item
-            // to execute the update in WriteJdbcP
+            // There is no input, we push the whole update/delete query to the database, but we need single dummy item
+            // to execute the update/delete in WriteJdbcP
             // We can consider refactoring the WriteJdbcP, so it doesn't need the dummy item in the future.
 
             // Use local member address to run the dummy source and processor on the same member
             Address localAddress = context.getNodeEngine().getThisAddress();
 
             Vertex v = dummySourceVertex(context, "DummySourceFor" + statement, localAddress);
-            Vertex updateVertex = context.getDag().newUniqueVertex(
+            Vertex dmlVertex = context.getDag().newUniqueVertex(
                     statement + "(" + table.getExternalNameList() + ")",
                     forceTotalParallelismOne(processorSupplier, localAddress)
             );
 
-            context.getDag().edge(Edge.between(v, updateVertex));
-            return updateVertex;
+            context.getDag().edge(Edge.between(v, dmlVertex));
+            return dmlVertex;
         } else {
-            Vertex updateVertex = context.getDag().newUniqueVertex(
+            Vertex dmlVertex = context.getDag().newUniqueVertex(
                     statement + "(" + table.getExternalNameList() + ")",
                     processorSupplier
             ).localParallelism(1);
-            return updateVertex;
+            return dmlVertex;
         }
     }
 

